src/test/java/com/coherent/task60/SingletoneWebDriver.java

```diff
@@ -4,18 +4,44 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
 
 public final class SingletoneWebDriver {
-    private static WebDriver driver;
+    private static RemoteWebDriver driver;
 
    private SingletoneWebDriver() {
     }
 
     public static WebDriver getDriver(){
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--remote-allow-origins=*");
+
+       //QUESTION: If I want to run tests on several browsers and operating systems: how can I effectively set up the tests?
```

Comment:
Create *.property file where you can have variable **environment**. It can have value _remote_ or _local_
In the code you you will read variable from property file and depending on the value different code will be executed. You can use  if-else construction (or switch-case) 
You can add as many variables to the file as you need. Values in the file will be changed manually